rs/registry/canister_data_provider/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+use candid::Principal;
+use ic_interfaces_registry::{
+    RegistryDataProvider, RegistryTransportRecord, ZERO_REGISTRY_VERSION,
+};
+use ic_nns_constants::REGISTRY_CANISTER_ID;
+use ic_registry_transport::pb::v1::RegistryDelta;
+use ic_registry_transport::{
+    deserialize_get_changes_since_response, serialize_get_changes_since_request,
+};
+use ic_stable_structures::memory_manager::VirtualMemory;
+use ic_stable_structures::storable::Bound;
+use ic_stable_structures::{DefaultMemoryImpl, StableBTreeMap, Storable};
+use ic_types::registry::RegistryDataProviderError;
+use ic_types::RegistryVersion;
+use itertools::Itertools;
+use std::borrow::Cow;
+use std::cmp::Ordering;
+use std::collections::HashSet;
+use std::marker::PhantomData;
+
+pub struct StorableRegistryValue(Option<Vec<u8>>);
+
+impl Storable for StorableRegistryValue {
+    fn to_bytes(&self) -> Cow<'_, [u8]> {
+        self.0.to_bytes()
+    }
+
+    fn from_bytes(bytes: Cow<'_, [u8]>) -> Self {
+        Self(Option::from_bytes(bytes))
+    }
+
+    const BOUND: Bound = Bound::Unbounded;
+}
+
+#[derive(Clone, Ord, PartialOrd, Eq, PartialEq, Default)]
+pub struct StorableRegistryKey {
+    version: u64,
+    key: String,
+}
+
+// This value is set as 2 times the max key size present in the registry
```

Comment:
Is the max key size limit == 100? or was this just based on observation of current data?

Reply:
This is just the max key of the current data